motionplan/tpspace/ptgGridSim.go

```diff
@@ -108,8 +109,32 @@ func (ptg *ptgGridSim) MaxDistance() float64 {
 	return ptg.refDist
 }
 
-func (ptg *ptgGridSim) Trajectory(alpha, dist, resolution float64) ([]*TrajNode, error) {
-	return ComputePTG(ptg, alpha, dist, resolution)
+func (ptg *ptgGridSim) Trajectory(alpha, start, end, resolution float64) ([]*TrajNode, error) {
+	if math.Abs(start) > math.Abs(end) {
+		return nil, fmt.Errorf("cannot calculate trajectory, start %d cannot be greater than end %d", start, end)
+	}
+	if end == 0 {
+		return ComputePTG(ptg, alpha, end, resolution)
+	}
+	traj, err := ComputePTG(ptg, alpha, end, resolution)
+	if err != nil {
+		return nil, err
+	}
+	if start > 0 {
+		firstNode, err := computePTGNode(ptg, alpha, start)
+		if err != nil {
+			return nil, err
+		}
+		first := -1
```

Comment:
This probably never stays -1 but if it does it will segfault

Reply:
I'll add an error check